tests/conftest.py

```diff
@@ -2,49 +2,83 @@
 
 import os
 import sys
+import tempfile
 
 sys.path.append(os.path.join(os.path.dirname(__file__), "helpers"))
 
 import pytest
+
 from testplan import TestplanMock
-from testplan.common.utils import path
+from testplan.common.utils.path import default_runpath, makedirs
+
+
+# Testplan and various drivers have a `runpath` attribute in their config
+# and intermediate files will be placed under that path during running.
+# In PyTest we can invoke fixtures such as `runpath`, `runpath_module` and
+# `mockplan` so that a temporary `runpath` is generated. In order to easily
+# collect the output an environment variable `TEST_ROOT_RUNPATH` can be set,
+# it will be the parent directory of all those temporary `runpath`.
+
+
+def _generate_runpath():
+    """
+    Generate a temporary directory unless specified by environment variable.
+    """
+    if os.environ.get("TEST_ROOT_RUNPATH"):
+        yield tempfile.mkdtemp(dir=os.environ["TEST_ROOT_RUNPATH"])
+    else:
+        parent_runpath = default_runpath()
```

Comment:
I think it's fine to just use tempfile behavior if TEST_ROOT_RUNPATH is not specified. default_runpath() isn't meant to be used here.

Reply:
recoverd